src/Polly.Core/Simmy/MonkeyStrategy.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Polly.Simmy.Utils;
+
+namespace Polly.Simmy;
+
+#pragma warning disable CA1031 // Do not catch general exception types
+#pragma warning disable S3928 // Custom ArgumentNullException message
+
+/// <summary>
+/// Contains common functionality for chaos strategies which intentionally disrupt executions - which monkey around with calls.
+/// </summary>
+public abstract class MonkeyStrategy : ResilienceStrategy
+{
+    private readonly Func<double> _randomizer;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MonkeyStrategy"/> class.
+    /// </summary>
+    /// <param name="options">The chaos strategy options.</param>
+    protected MonkeyStrategy(MonkeyStrategyOptions options)
+    {
+        Guard.NotNull(options);
+        Guard.NotNull(options.Randomizer);
+
+        if (options.InjectionRate is null && options.InjectionRateGenerator is null)
+        {
+            throw new ArgumentNullException(nameof(options.InjectionRate), "Either InjectionRate or InjectionRateGenerator is required.");
+        }
+
+        if (options.Enabled is null && options.EnabledGenerator is null)
+        {
+            throw new ArgumentNullException(nameof(options.Enabled), "Either Enabled or EnabledGenerator is required.");
+        }
+
+        _randomizer = options.Randomizer;
+        InjectionRateGenerator = options.InjectionRate.HasValue ? (_) => new(options.InjectionRate.Value) : options.InjectionRateGenerator;
+        EnabledGenerator = options.Enabled.HasValue ? (_) => new(options.Enabled.Value) : options.EnabledGenerator;
+    }
+
+    /// <summary>
+    /// Gets the injection rate for a given execution, which the value should be between [0, 1].
+    /// </summary>
+    public Func<ResilienceContext, ValueTask<double>> InjectionRateGenerator { get; }
```

Comment:
```suggestion
    internal Func<ResilienceContext, ValueTask<double>>? InjectionRateGenerator { get; }
```

Two notes:

- Does it need to be public?
- Can we make it nullable? (since it's nullable on options)

Reply:
No need to be public but might be useful, and that's always set in the constructor so it's not gonna be null.